File: java/t/test/Structural/TestFacade.java
//TestFacade.java - testing the Facade


package test.Structural;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Structural.Facade.*;




class TestFacade {

	@Test
   public void facade () {
       FacadeCuppaMaker cuppaMaker = new FacadeCuppaMaker();
       FacadeTeaCup teaCup = cuppaMaker.makeACuppa();
		
       org.junit.Assert.assertEquals("Facade test", 
			"A nice cuppa!",
			teaCup.toString()
		);
   }
}

File: csharp/src/Structural/Facade/TestFacade.cs
//TestFacade.java - testing the Facade


namespace Structural.Facade {

using tap;


class TestFacade {
   public static void Main() {
       FacadeCuppaMaker cuppaMaker = new FacadeCuppaMaker();
       FacadeTeaCup teaCup = cuppaMaker.makeACuppa();
		Tapper tap = new Tapper();
       tap.test("Facade test", 
			teaCup.ToString() , ("A nice cuppa tea!"));
		tap.done();
   }
}

}
